src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipPath.java

```diff
@@ -200,13 +200,18 @@ public URI toUri() {
             return new URI("jar",
                            decodeUri(zfs.getZipFile().toUri().toString()) +
                            "!" +
-                           zfs.getString(toAbsolutePath().path),
+                           getRealPath(getResolvedPath()),
                            null);
         } catch (Exception ex) {
             throw new AssertionError(ex);
         }
     }
 
+    private String getRealPath(byte[] resolvedPath) {
+        byte[] path = zfs.lookupPath(resolvedPath);
```

Comment:
Can getRealPath be changed to be no-arg method that calls getResolvedPath and then does the lookup. I think that would be a clearer than assuming the argument is a resolved path.

Reply:
Done via https://github.com/openjdk/jdk/pull/6768/commits/12e106c6e69449f1fbde9e5b6a5ff3305c5de547